### DISCUSSION TITLE
How are license keys actually "used"?

### DISCUSSION THREAD

**Discussion by hambabler** [keygen-sh/community#128] [Q&A] [answered]
I started the smaller plan on keygen just to test it out, and I noticed that I could activate the machine with said license key but no user actually ends up being tied to this license key. I tried using increment-usage but that is only available in server-sided operations, but im trying to make most of the use license activation and sign-up process client sided. Is there a direct request I can send to have a user "use" a license key?

**Comment by ezekg** [comment]
You can increment a license's usage client-side using [license authentication](https://keygen.sh/docs/api/authentication/#license-authentication). I don't understand what you mean by "no user is tied to the license." Can you explain your use case and your expectations? If you want to assign a user to a license, you need to create a user, and then assign the user to the license during license creation. But please explain your use case and I can point you in the right direction.

**Reply by hambabler** [reply]
Sorry if I wasn't clear. In my application, I'm trying to let the user activate an unused license so it connects to their user object. I assume I would need to set the user relationship for the license client sided?

**Reply by ezekg** [reply, marked as answer]
We do not currently support assigning users to licenses client-side, because not all licenses are meant to have users. This would likely need to be done server-side. You may be able to work around this by using machine activation to activate a "user" by using an email as the machine fingerprint. But actually assigning a user to an unassigned license client-side is not something currently on our roadmap. But I can make a ticket for you that we may explore in the future.
